### PR TITLE
Extract swagger documentation to separate module

### PR DESCRIPTION
**Pull request by doncem**
### Change description ###

Extract swagger documentation to separate module. Affected changes:
- get rid of swagger incurred extra libraries from main application
- docker runs same as usual just building mirror of application including swagger configuration

Cons:
- Would be good to use separate port. Could not make it work
- Would be better to extract port to (environment) properties. Could not make it work for swagger module - still kept on using main application value instead of overridden one

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by rkondratowicz** [comment]
This also stops the app from serving swagger spec when it's running.
Is that what we want @jasonpaige ?

**Comment by jasonpaige** [comment]
I'm assuming in dev you'd probably still want to be able to run the swagger spec?

File: build.gradle [hunk 1]
```diff
@@ -70,12 +76,13 @@ dependencies {
...
   compile group: 'org.flywaydb', name: 'flyway-core', version: '4.1.2'
 
-  compile group: 'io.springfox', name: 'springfox-swagger2', version: versions.springfoxSwagger
-  compile group: 'io.springfox', name: 'springfox-swagger-ui', version: versions.springfoxSwagger
+  compileOnly group: 'io.swagger', name: 'swagger-annotations', version: '1.5.17'
 
   compile group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: versions.jackson
   compile group: 'com.fasterxml.jackson.datatype', name: 'jackson-datatype-jsr310', version: versions.jackson
 
+  compile group: 'com.google.guava', name: 'guava', version: '20.0'
```
Comment (timja-kainos) [inline build.gradle]: what is guava needed for here?

Comment (doncem) [reply build.gradle]: for [CryptoService](https://github.com/hmcts/draft-store/blob/master/src/main/java/uk/gov/hmcts/reform/draftstore/service/crypto/CryptoService.java)

**Comment by doncem** [comment]
Can boot up `docker-compose up draft-store-swagger` and will have all the previous endpoints present whereas `docker-compose up draft-store-api` - all but swagger

**Comment by jasonpaige** [comment]
that approach works for me as long as it's documented. How about others?

**Comment by jasonpaige** [comment]
As for the maintainer to cover codacy, just suppress the error on the project for now (need to login to codacy i think)

**Comment by rkondratowicz** [comment]
is it worth it? do we want to change all the spring services now?
Are we not overcomplicating it?

File: build.gradle (hunk 1 above)
Comment (rkondratowicz) [reply build.gradle]: and it previously wasn't? O_o

**Comment by timja-kainos** [comment]
also not sure if its worth it

File: build.gradle (hunk 1 above)
Comment (doncem) [reply build.gradle]: beats me. only can assume was injected together with springfox - did not check maven repo for dependencies, but guava pom has smart plugin injection and cannot **just** grab `google.common` packages ¯\\\_(ツ)_/¯

if it is true (about springfox) this is why it's not great to have such libraries in main source code just for swagger docs. docs in general could be separated

**Comment by doncem** [comment]
Benefit: no extra libs in source code
"Worthness": depends on the view. can easily scratch it to 'none' - why spend time separating when it all works as is.

When it comes down to service / source code quality, I'd rather keep away such _extra_ code separate to minify the main project to bare bones ensuring everything is actually needed for production. As expected here - as soon as moved away swagger libraries - crypto service stopped working. Or might be some local issue as it was the first time I compiled draft store

We have centralised api spec place which is generated during CI process. But if we agree on leaving as it was - can just close the PR. No fuss

P.S. Jason, it's your app Codacy report ran against :D

**Comment by doncem** [comment]
Closing according to failure of worthiness and lack of documented proof how/whether reform microservices should serve API documentation.

**Pull request closed without merging**